x/mongo/driver/topology/errors.go

```diff
@@ -95,14 +102,19 @@ func (w WaitQueueTimeoutError) Error() string {
 		)
 	}
 
-	return fmt.Sprintf(
-		"%s; maxPoolSize: %d, connections in use by cursors: %d"+
-			", connections in use by transactions: %d, connections in use by other operations: %d",
-		errorMsg,
-		w.maxPoolSize,
-		w.PinnedCursorConnections,
-		w.PinnedTransactionConnections,
-		uint64(w.totalConnectionCount)-w.PinnedCursorConnections-w.PinnedTransactionConnections)
+	msg := fmt.Sprintf("%s; maxPoolSize: %d, ", errorMsg, w.maxPoolSize)
+	if pinnedConnections := w.pinnedConnections; pinnedConnections != nil {
+		openConnectionCount := uint64(w.totalConnectionCount) -
+			(*pinnedConnections).cursorConnections -
+			(*pinnedConnections).transactionConnections
+		msg += fmt.Sprintf("connections in use by cursors: %d, connections in use by transactions: %d, connections in use by other operations: %d, ",
```

Comment:
Optional: I've generally found the "connections in use by other operations" to be confusing and not very useful. The things we most commonly want to know are:
1. Is the connection pool saturated (i.e. is `totalConnectionCount == maxPoolSize`)?
2. Are there any idle connections?

You can derive that by using "connections in use by other operations", but a much simpler way to answer the above questions is to just include open connections and idle connections counts. Also, we should include the total connection count independent of if there are any pinned connections.

Consider removing the "connections in use by other operations" count and replacing it with an "open connections" count that is always part of the error message. I see that idle connection count is already part of the error message 👍

Reply:
I'd like to keep "connections in use by other operations" now as it's required in the [specs](https://github.com/mongodb/specifications/blob/master/source/load-balancers/load-balancers.rst#connection-tracking) and its [test case](https://github.com/mongodb/specifications/blob/master/source/load-balancers/tests/wait-queue-timeouts.json#L87). Maybe, we can create a specs ticket to update the message?

Reply:
Sounds good 👍